### PR TITLE
[WIP][LG] achieve DCR #2055,Unify the interface of TemplateEngine across f…

### PR DESCRIPTION
**Pull request by Danieladu**
[LG] achieve DCR #2055,Unify the interface of TemplateEngine across file and text
#2055 

```
new TemplateEngine().AddText(string content, string name, ImportResolverDelegate importResolver)
new TemplateEngine().AddFile(string filepath,ImportResolverDelegate importResolver)
new TemplateEngine().AddFiles(IEnumerable<string> filepaths, ImportResolverDelegate importResolver)
```

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs [hunk 1]
```diff
@@ -85,6 +59,13 @@ public TemplateEngine Add(string[] filePaths)
             return this;
         }
 
+        /// <summary>
+        /// Load single .lg file into template engine.
+        /// </summary>
+        /// <param name="filePath">Paths to .lg file.</param>
+        /// <returns>Teamplate engine with single parsed file.</returns>
+        public TemplateEngine Add(string filePath) => Add(new List<string> { filePath });
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs]: >Add [](start = 30, length = 3)

I think I would name this AddFile()

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs [hunk 2]
```diff
@@ -70,7 +44,7 @@ public TemplateEngine()
         /// </summary>
         /// <param name="filePaths">Paths to .lg files.</param>
         /// <returns>Teamplate engine with parsed files.</returns>
-        public TemplateEngine Add(string[] filePaths)
+        public TemplateEngine Add(IEnumerable<string> filePaths)
```
Comment (tomlm) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs]: >Add [](start = 30, length = 3)

Rename this to AddFiles()

**Comment by tomlm** [comment]
I'm down with this, but I think it's better to have explicit AddFile/AddFiles() methods, .Add() is too overloaded.

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs (hunk 1 above)
Comment (Danieladu) [reply libraries/Microsoft.Bot.Builder.LanguageGeneration/TemplateEngine.cs]: It makes sense

**Comment by boydc2014** [comment]
@Danieladu rename to AddFile/AddFiles make sense, and we want those two functions also support delegate.

**Comment by Danieladu** [comment]
> @Danieladu rename to AddFile/AddFiles make sense, and we want those two functions also support delegate.

sure

**Pull request merged**